temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowRunTaskHandler.java

```diff
@@ -363,4 +370,41 @@ public void cancel(HistoryEvent cancelEvent) {
       replayWorkflowExecutor.handleWorkflowExecutionCancelRequested(cancelEvent);
     }
   }
+
+  private static class LocalActivityMeteringHelper {
+    private final Map<String, AtomicInteger> firstWftActivities = new HashMap<>();
+    private final Map<String, AtomicInteger> nonFirstWftActivities = new HashMap<>();
+    private final Set<String> completed = new HashSet<>();
+
+    private void newWFTStarting() {
+      for (String activityId : firstWftActivities.keySet()) {
+        AtomicInteger removed = firstWftActivities.remove(activityId);
+        removed.set(0);
+        nonFirstWftActivities.put(activityId, removed);
+      }
+    }
+
+    private void addNewLocalActivity(ExecuteLocalActivityParameters params) {
+      AtomicInteger attemptsDuringWFTCounter = new AtomicInteger(0);
+      params.setOnNewAttemptCallback(attemptsDuringWFTCounter::incrementAndGet);
```

Comment:
Eh, I think since you end up having to pass both the id and the atomic int itself, it ends up being kinda messy - if the class weren't already nested I think I'd agree but the two are closely associated in a way I think makes sense.